polaris-react/src/components/Badge/Badge.scss

```diff
@@ -11,29 +13,57 @@
   border-radius: var(--p-border-radius-5);
   color: var(--p-color-text);
 
+  :where(#{$se23}) & {
+    // stylelint-disable-next-line -- se23 bg-transparent-hover
+    background-color: var(--p-color-experimental-bg-transparent-subdued);
+    color: var(--p-color-text-subdued);
```

Comment:
you might have to add `font-weight: var(--p-font-weight-medium)` here it looks like they are using a new weight on the badges in Figma
 
<img width="76" alt="Screenshot 2023-05-26 at 3 13 23 PM" src="https://github.com/Shopify/polaris/assets/20652326/551a41ae-f975-4f47-9a7a-e0afb9d5c448">
<img width="76" alt="Screenshot 2023-05-26 at 3 13 35 PM" src="https://github.com/Shopify/polaris/assets/20652326/46001e10-6b52-41ad-895d-7f2989013817">

Reply:
not sure if this is best done in scss or directly in the `Text` component but either works